platform/commonUI/edit/src/capabilities/TransactionalPersistenceCapability.js

```diff
@@ -67,10 +67,17 @@ define(
             }
 
             function onCancel() {
-                return self.persistenceCapability.refresh().then(function (result) {
+                if (self.domainObject.getModel().persisted) {
```

Comment:
Hehehe, you know I did actually think when I reviewed the diff on creating the PR that this introduces a potential bug, but figured we wouldn't have objects that were persisted at midnight on January 1st, 1970 so I left it :) But you're totally right, and it's not even a matter of pedantry the code is just wrong. I'll fix it.